ClassAttendance/src/WebUI/Facades/StudentsFacade.cs

```diff
@@ -1,26 +1,36 @@
 ﻿using BLL.Interfaces;
 using BLL.Models;
+using RestEase;
 using System.Collections.Generic;
 using System.Linq;
+using System.Net.Http;
 using System.Threading.Tasks;
+using WebUI.Api;
 using WebUI.Models.ViewModels.StudentModel;
 
 namespace WebUI.Facades
 {
     public class StudentsFacade
     {
-        private readonly IStudentService _studentService;
+        private const string ClientName = "StudentApi";
+        private readonly IStudentApi _studentApi;
         private readonly IService<Group> _groupService;
 
-        public StudentsFacade(IStudentService studentService, IService<Group> groupService)
+        public StudentsFacade(IHttpClientFactory httpClientFactory, IService<Group> groupService)
         {
-            _studentService = studentService;
+            if (httpClientFactory is null)
+            {
+                throw new System.ArgumentNullException(nameof(httpClientFactory));
+            }
+
+            var client = httpClientFactory.CreateClient(ClientName);
```

Comment:
From my perspective - it would be better to register IStudentApi in Startup.cs like this:
```csharp
services.AddHttpClient("name")
.ConfigureHttpClient(c => c.BaseAddress = new Uri(baseUrl))
.AddTypedClient(RestClient.For<IStudentApi>);
```

Reply:
Didn't know i can do that this way

Reply:
You can.

Reply:
It's practical example.